lib/acts_as_tenant/model_extensions.rb

```diff
@@ -103,21 +87,19 @@ def self.included(base)
 
     module ClassMethods
       def acts_as_tenant(tenant = :account, options = {})
-        ActsAsTenant.set_tenant_klass(tenant)
-
         ActsAsTenant.add_global_record_model(self) if options[:has_global_records]
 
         # Create the association
         valid_options = options.slice(:foreign_key, :class_name, :inverse_of, :optional)
-        fkey = valid_options[:foreign_key] || ActsAsTenant.fkey
-        polymorphic_type = valid_options[:foreign_type] || ActsAsTenant.polymorphic_type
+        fkey = valid_options[:foreign_key] || "#{tenant}_id".to_sym
+        polymorphic_type = valid_options[:foreign_type] || "#{tenant}_type"
```

Comment:
lol wut. how is `valid_options[:foreign_type]` supposed to ever return truthy if `valid_options` will never have a `:foreign_type`

Reply:
You're right. Yet another problem with polymorphic tenants. Note that this is existing code. Should add a fix for this to https://github.com/ErwinM/acts_as_tenant/pull/190